frontend/src/lib/utils/format-number.ts

```diff
@@ -1,12 +1,122 @@
-export function formatNumber(value: number): string {
-  if (value >= 1_000_000_000) {
-    return `${(value / 1_000_000_000).toFixed(2)}B`;
+/**
+ * Formats a price with a maximum of 4 leading zeros normally, using subscript notation for more zeros
+ * @param value The price value to format
+ * @returns Formatted price string (e.g., "$1234", "$0.0002525", "$0.0₄2525")
+ */
+export function formatPrice(value: number): string {
```

Comment:
I believe you can use this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat
MINOR